anchore_engine/analyzers/binary.py

```diff
@@ -1,18 +1,133 @@
+import collections
+
+import anchore_engine.utils
+from anchore_engine.subsys import logger
+from . import utils
+
+binary_package_el = {
+    "name": None,
+    "version": None,
+    "location": None,
+    "type": "binary",
+    "files": [],
+    "license": "N/A",
+    "origin": "N/A",
+    "metadata": {},
+}
+
+
+# this is a direct port of the binary analyzer module to here... this should be refactored
+def catalog_image(allpkgfiles, unpackdir):
+    squashtar = os.path.join(unpackdir, "squashed.tar")
+
+    # get a listing of all files from either a previous run or the squashtar
+    allfiles = {}
+    if os.path.exists(unpackdir + "/anchore_allfiles.json"):
+        with open(unpackdir + "/anchore_allfiles.json", "r") as FH:
+            allfiles = json.loads(FH.read())
+    else:
+        fmap, allfiles = utils.get_files_from_squashtar(
+            os.path.join(unpackdir, "squashed.tar")
+        )
+        with open(unpackdir + "/anchore_allfiles.json", "w") as OFH:
+            OFH.write(json.dumps(allfiles))
+
+    # set up ordered dictionary structure for the runtimes and evidence types
+    evidence = collections.OrderedDict()
+    for runtime in ["python", "go", "busybox"]:
+        evidence[runtime] = collections.OrderedDict()
+        for etype in ["binary", "devel"]:
+            evidence[runtime][etype] = []
+
+    # perform a per file routine to evaluate files for gathering binary package version evidence
+    with tarfile.open(
+        os.path.join(unpackdir, "squashed.tar"), mode="r", format=tarfile.PAX_FORMAT
+    ) as tfl:
+        alltnames = tfl.getnames()
+        alltfiles = {}
+        for name in alltnames:
+            alltfiles[name] = True
+
+        memberhash = utils.get_memberhash(tfl)
+        for member in list(memberhash.values()):
+            try:
+                _get_python_evidence(tfl, member, memberhash, evidence)
+            except Exception as err:
+                logger.exception(
+                    "Unexpected exception evaluating file='{}' for python runtime evidence".format(
+                        member.name
+                    )
+                )
+
+            try:
+                _get_golang_evidence(tfl, member, memberhash, evidence)
+            except Exception as err:
+                logger.exception(
+                    "Unexpected exception evaluating file='{}' for python golang evidence".format(
+                        member.name
+                    )
+                )
+
+            try:
+                _get_busybox_evidence(tfl, member, memberhash, evidence)
+            except Exception as err:
+                logger.exception(
+                    "Unexpected exception evaluating file='{}' for python busybox evidence".format(
+                        member.name
+                    )
+                )
+
+    # write evidence back to a results data structure
+    resultlist = {}
+    version_found_map = {}
+    for runtime in evidence.keys():
```

Comment:
there is no need to do `.keys()` here, it is the default when looping over dictionaries, same suggestion on the below loops

Reply:
Actually, I think I am missing the fact that a lot of this is copy/pasta, feel free to ignore :)